src/main/java/org/commcare/xml/EndpointParser.java

```diff
@@ -0,0 +1,48 @@
+package org.commcare.xml;
+
+import org.commcare.suite.model.Endpoint;
+import org.commcare.suite.model.StackOperation;
+import org.javarosa.xml.ElementParser;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+import java.util.Vector;
+
+public class EndpointParser extends ElementParser<Endpoint> {
+
+    static final String NAME_ENDPOINT = "endpoint";
+    private static final String NAME_ARGUMENT = "argument";
+    private static final String NAME_STACK = "stack";
+
+    public EndpointParser(KXmlParser parser) {
+        super(parser);
+    }
+
+    @Override
+    public Endpoint parse() throws InvalidStructureException, IOException, XmlPullParserException, UnfullfilledRequirementsException {
+        String id = parser.getAttributeValue(null, "id");
```

Comment:
Can this `id` be null or empty?

Reply:
it should not be. I should add a validation here to that point. Thanks.